mainloop.h

```diff
@@ -19,21 +19,22 @@
 
 #include "comm.h"
 #include "endpoint.h"
+#include "router.h"
 
 struct endpoint_entry {
     struct endpoint_entry *next;
     Endpoint *endpoint;
     bool remove;
 };
 
-class Mainloop {
+class Mainloop : public Router {
```

Comment:
I don't think that we will need another Router implementation, Mainloop is only implementing Router because this way Endpoint can include router.h without cause a circular dependency.